### PR TITLE
Remove 2.7 & 3.4, add 3.8 to supported versions

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [X] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?
* Add Python 3.8 to the list of supported versions
* Remove Python 2.7 and 3.4 support
    * `setup.py` - used by PyPI
    * `_mock_helper.py` - string/byte support
    * functions
    * Readme
### ~~What testing has been done?~~

### CODE REVIEW

**Comment by texasaggie97** [comment]
test this please

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,8 +23,10 @@ All notable changes to this project will be documented in this file.
 ## [Unreleased]
 * ### ALL
     * #### Added
+        * Support for Python 3.8
```
Comment (marcoskirsch) [inline CHANGELOG.md]: Didn't we do this earlier? I suppose now it's "official"?

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,8 +23,10 @@ All notable changes to this project will be documented in this file.
...
     * #### Changed
     * #### Removed
+        * Support for Python 2.7 and 3.4
```
Comment (marcoskirsch) [inline CHANGELOG.md]: Add something like: These versions of Python are no longer supported by the Python Software Foundation, and a link to the support doc.

File: README.rst [hunk 3]
```diff
@@ -59,7 +59,7 @@ It is implemented as a set of `Mako templates <http://makotemplates.org>`_ and p
 
 **nimi-python** supports all the Operating Systems supported by the underlying driver.
 
-**nimi-python** supports Python 2.7, 3.4 and later using CPython or PyPy.
+**nimi-python** follows Python Organization's life cycle plan. Current versions supported: Python 3.5 and later using CPython or PyPy.
```
Comment (marcoskirsch) [inline README.rst]: "Python Software Foundation" is the correct name. And link to doc.

File: src/niscope/system_tests/test_system_niscope.py [hunk 4]
```diff
@@ -87,14 +87,11 @@ def test_fetch_binary8_into(session):
...
     for i in range(len(waveforms)):
-        if sys.version_info.major >= 3:
-            # Only python 3 will have the record memory view in the wfm_info
-            record_wfm = waveforms[i].samples
-            assert len(record_wfm) == test_record_length
-            for j in range(len(record_wfm)):
-                assert record_wfm[j] == waveform[i * test_record_length + j]
-        else:
-            assert waveforms[i].samples is None
+        # Only python 3 will have the record memory view in the wfm_info
```
Comment (marcoskirsch) [inline src/niscope/system_tests/test_system_niscope.py]: remove comment

File: src/niscope/system_tests/test_system_niscope.py [hunk 5]
```diff
@@ -115,14 +112,11 @@ def test_fetch_binary16_into(session):
...
```
Comment (marcoskirsch) [inline src/niscope/system_tests/test_system_niscope.py]: remove

File: src/niscope/system_tests/test_system_niscope.py [hunk 6]
```diff
@@ -171,14 +162,11 @@ def test_fetch_double_into(session):
...
```
Comment (marcoskirsch) [inline src/niscope/system_tests/test_system_niscope.py]: remove

File: CHANGELOG.md (hunk 1 above)
Comment (texasaggie97) [reply CHANGELOG.md]: Yes, until the issue with scipy was fixed (it is now) we didn't say we officially supported 3.8

File: CHANGELOG.md (hunk 2 above)
Comment (texasaggie97) [reply CHANGELOG.md]: Updated

File: README.rst (hunk 3 above)
Comment (texasaggie97) [reply README.rst]: Blurb now states:
**nimi-python** follows `Python Software Foundation <https://devguide.python.org/#status-of-python-branches>`_ support policy for different versions. At this time this includes Python 3.5 and above using CPython or PyPy.

File: src/niscope/system_tests/test_system_niscope.py (hunk 4 above)
Comment (texasaggie97) [reply src/niscope/system_tests/test_system_niscope.py]: Removed

File: src/niscope/system_tests/test_system_niscope.py (hunk 5 above)
Comment (texasaggie97) [reply src/niscope/system_tests/test_system_niscope.py]: Removed

File: src/niscope/system_tests/test_system_niscope.py (hunk 6 above)
Comment (texasaggie97) [reply src/niscope/system_tests/test_system_niscope.py]: Removed

File: README.rst [hunk 7]
```diff
@@ -42,7 +42,8 @@ Author       National Instruments
 About
 =====
 
-The **nimi-python** repository generates Python bindings (Application Programming Interface) for interacting with the Modular Instrument drivers. Currently, the following drivers are supported:
+The **nimi-python** repository generates Python bindings (Application Programming Interface) for interacting with the Modular Instrument drivers. Currently,
```
Comment (marcoskirsch) [inline README.rst]: I'd change the last sentece to say:

"The following driver APIs are supported" (add API, remove Currently)

File: docs/_static/about.inc [hunk 8]
```diff
@@ -3,7 +3,8 @@
 About
 =====
 
-The **nimi-python** repository generates Python bindings (Application Programming Interface) for interacting with the Modular Instrument drivers. Currently, the following drivers are supported:
+The **nimi-python** repository generates Python bindings (Application Programming Interface) for interacting with the Modular Instrument drivers. Currently,
```
Comment (marcoskirsch) [inline docs/_static/about.inc]: This text is duplicated from README.rst.
We don't single source this text, right? I think that's fine just looking for confirmation.

File: docs/conf.py [hunk 9]
```diff
@@ -55,7 +55,7 @@
 
 # General information about the project.
 project = 'NI Modular Instruments Python API'
-copyright = '2017-2019, National Instruments'
+copyright = '2017-2020, National Instruments'
```
Comment (marcoskirsch) [inline docs/conf.py]: Should we programmatically fill this out? If so, I can open an issue for it.

File: src/niscope/system_tests/test_system_niscope.py [hunk 10]
```diff
@@ -143,14 +135,11 @@ def test_fetch_binary32_into(session):
...
```
Comment (marcoskirsch) [inline src/niscope/system_tests/test_system_niscope.py]: This comment is outdated.

File: docs/_static/about.inc (hunk 8 above)
Comment (texasaggie97) [reply docs/_static/about.inc]: This is the single source for all the different locations including the global README.rst

File: README.rst (hunk 7 above)
Comment (texasaggie97) [reply README.rst]: Changed

File: docs/conf.py (hunk 9 above)
Comment (texasaggie97) [reply docs/conf.py]: It is programmatically set along with the version. Look at build/templates/conf.py.mako

File: src/niscope/system_tests/test_system_niscope.py (hunk 10 above)
Comment (texasaggie97) [reply src/niscope/system_tests/test_system_niscope.py]: Fixed

**Comment by texasaggie97** [comment]
test this please

**Pull request merged**